### ISSUE TITLE
Use data-testid instead of aria-label for test selectors

### ISSUE THREAD

**Issue by joshhunt** [grafana/grafana#36523]
Grafana [currently recommends](https://github.com/grafana/grafana/blob/main/contribute/style-guides/e2e.md) using the aria-label attribute as a way to both describe elements in the UI and create selectors for using in unit and E2E tests. As the Grafana team has scaled, this is no longer a good recommendation because it ends up hampering accessibility by using _too many_ label that might not be that useful, or may be out of date. It also makes translating Grafana difficult.

As we've discussed internally, we would like to switch away from using aria-labels in tests and instead use the `data-testid` attribute where other more user-centric selectors (such as actual labels) are not available. 

It's anticipated that _most_ aria-labels in the codebase would be removed, but care would need to be taken to keep existing aria-labels that are there for genuine accessibility reasons there. I think those labels should be moved to be inline, just like visible labels on the page.

Tasks

- [x] Deprecate aria-label usage in grafana-e2e selector library
- [ ] Support data-testid in grafana-e2e selector library as the "primary" selector
- [ ] Migrate Grafana codebase to use data-testid for tests, leaving aria-labels in where needed for screen readers.
```[tasklist]
### Tasks
- [ ] https://github.com/grafana/grafana/issues/78408
- [ ] https://github.com/grafana/grafana/issues/78409
- [ ] https://github.com/grafana/grafana/issues/78410
- [ ] https://github.com/grafana/grafana/issues/79609
- [ ] https://github.com/grafana/grafana/issues/78411
- [ ] https://github.com/grafana/grafana/issues/78412
- [ ] https://github.com/grafana/grafana/issues/78413
- [ ] https://github.com/grafana/grafana/issues/78414
- [ ] https://github.com/grafana/grafana/issues/78416
```

**Comment by joshhunt** [comment]
To migrate your code from using aria-label to data-testid for e2e selectors, check out this PR https://github.com/grafana/grafana/pull/78399. Essentially:

**Replace any aria-label props using values from e2e-selectors with data-testid:**

```diff
<section
-   aria-label={selectors.components.TimeZonePicker.container}
+   data-testid={selectors.components.TimeZonePicker.container}
    className={cx(style.timeZoneContainer, style.timeSettingContainer)}
 >
```

**Update the selector value, prefixing it with `data-testid`.** This ensures that the e2e framework selects based on data-testid instead of aria-label

```diff
TimeZonePicker: {
-   container: "time zone picker container"
+   container: "data-testid time zone picker container"
}
```

**While you're there, assess the component and make sure it actually has a label**

For example, prefer using `IconButton` with a `tooltip` prop:

```diff
-<Button
-   aria-label={selectors.components.TimePicker.calendar.closeButton}
-   icon="times"
-   onClick={onClose}
-/>
+<IconButton
+   data-testid={selectors.components.TimePicker.calendar.closeButton}
+   tooltip={t(`time-picker.calendar.close`, 'Close calendar')}
+   name="times"
+   onClick={onClose}
+/>
```

**Or, if `IconButton` is unsuitable, give it an explicit `aria-label`** for people, rather than machines

```diff
<Button
-   aria-label={selectors.components.TimePicker.calendar.openButton}
+   data-testid={selectors.components.TimePicker.calendar.openButton}
+   aria-label={t('time-picker.range-content.open-input-calendar', 'Open calendar')}
```

**Input fields should also have explicit labels:**

```diff
+const fromFieldId = useId() // import { fromFieldId } from "react"

<Field label={t('time-picker.range-content.from-input', 'From')}>
    <Input
+       id={fromFieldId}
-       aria-label={selectors.components.TimePicker.fromField}
+       data-testid={selectors.components.TimePicker.fromField}
        value={from.value}
        onChange={(event) => onChange(event.currentTarget.value, to.value)}
    />
</Field>
```

**Prefer using human-readable labels in jest tests over e2e selectors**

```diff
-expect(getByLabelText(selectors.components.TimePicker.fromField)).toBeInTheDocument();
+expect(getByLabelText('From')).toBeInTheDocument();
```

**Comment by mckn** [comment]
Thanks for opening this issue @joshhunt and great work with the migration guide 🙏🏻

**Comment by joshhunt** [comment]
Will close when we move lint rule from betterer to eslint proper. Will probably exempt all of old dashboards stuff that will be removed soon.
